processing/src/main/java/org/apache/druid/segment/vector/VectorColumnSelectorFactory.java

```diff
@@ -62,16 +66,29 @@ default int getMaxVectorSize()
   SingleValueDimensionVectorSelector makeSingleValueDimensionSelector(DimensionSpec dimensionSpec);
 
   /**
-   * Returns a string-typed, multi-value-per-row column selector. Should only be called on columns where
-   * {@link #getColumnCapabilities} indicates they return STRING. Unlike {@link #makeSingleValueDimensionSelector},
-   * this should not be called on nonexistent columns.
+   * Returns a dictionary encoded, string-typed, multi-value-per-row column selector. Should only be called on columns
+   * where {@link #getColumnCapabilities} indicates they return STRING. Unlike
+   * {@link #makeSingleValueDimensionSelector}, this should not be called on nonexistent columns.
    *
    * If you need to write code that adapts to different input types, you should write a
    * {@link org.apache.druid.segment.VectorColumnProcessorFactory} and use one of the
    * {@link org.apache.druid.segment.ColumnProcessors#makeVectorProcessor} functions instead of using this method.
    */
   MultiValueDimensionVectorSelector makeMultiValueDimensionSelector(DimensionSpec dimensionSpec);
 
+  /**
+   * Returns a dimension object selector. Useful for non-dictionary encoded strings, or when using the dictionary
+   * isn't useful. Right now, this should probably only be called on single valued STRING inputs (multi-value STRING
+   * vector object selector is not yet implemented), but perhaps a world can be imagined where this could work with
+   * COMPLEX too perhaps if it were useful for them to interact with {@link DimensionSpec} in some manner? May be called
+   * for non-existent columns.
```

Comment:
These javadocs should explicitly constrain the type of Object that may show up in the returned VectorObjectSelector. Is it always going to be a String or a null? Might it be `String[]`, if the underlying column is multi-value?